implementation/src/main/java/io/smallrye/mutiny/groups/UniOnNull.java

```diff
@@ -88,26 +68,7 @@ public UniOnNull(Uni<T> upstream) {
      */
     public Uni<T> switchTo(Supplier<Uni<? extends T>> supplier) {
         Supplier<Uni<? extends T>> actual = Infrastructure.decorate(nonNull(supplier, "supplier"));
-
-        Uni<T> uni = upstream.onItem().transformToUni(res -> {
-            if (res != null) {
-                return Uni.createFrom().item(res);
-            } else {
-                Uni<? extends T> produced;
-                try {
-                    produced = actual.get();
-                } catch (Throwable e) {
-                    return Uni.createFrom().failure(e);
-                }
-
-                if (produced == null) {
-                    return Uni.createFrom().failure(new NullPointerException(SUPPLIER_PRODUCED_NULL));
```

Comment:
Not too sure about this one. I didn't want to include `continueWith()` and `switchTo()` in the API of `UniOnPredicate` because I think they are specific to the null case and therefore don't make much sense in that generalized case.

However, by trying to use `UniOnPredicate` also for these methods, we change the behavior of these methods slightly - for example, they now return `IllegalArgumentException` instead of `NullPointerException`. And especially in the `continueWith()` case we now don't have the check anymore that the result of the `supplier` is not null, which is why I also had to delete one test case.

Do you think this behavior change is okay and favorable to have all of these methods use the `UniOnPredicate`? Or should we keep the logic here in `UniOnNull` to keep the current behavior?

Reply:
It didn't occur to me earlier, but I might actually be able to both include these additional null checks while still using `UniOnPredicate`, since I could simply include the checks in the mapping functions. Then of course we would still have a little logic in `UniOnNull`, but not much. And we could keep the original behavior. Just let me know which solution you prefer.

Reply:
Changed it now to retain the old behavior of `continueWith()` and `switchTo()`. I think it's cleaner this way.